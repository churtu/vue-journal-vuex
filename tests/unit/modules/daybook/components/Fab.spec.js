import { shallowMount } from '@vue/test-utils'
import Fab from '@/modules/daybook/components/Fab';
;




describe( 'Fab component', () => {
    
    // test('', () => {
        
    // })

    test('Debe hacer match con el snapshot', () => {
        const wrapper = shallowMount( Fab )
        expect( wrapper.html() ).toMatchSnapshot()
    })
    
    test('Debe mostrar el icono por defecto', () => {
        const wrapper = shallowMount( Fab )
        const fab = wrapper.find( 'i' )
        expect( fab.classes( 'fa-plus' )).toBeTruthy()

    })

    test('Debe mostrar el icono por argumento: fa-circle', () => {
        const wrapper = shallowMount( Fab, {
            props: {
                icon: 'fa-circle'
            }
        })
        const fab = wrapper.find( 'i' )
        expect( fab.classes( 'fa-circle' )).toBeTruthy()
    })

    test('Debe emitir el evento on:click cuando se hace click en el boton', () => {
        const wrapper = shallowMount( Fab ) 
        const button = wrapper.find('button')
        button.trigger('click')
        // console.log(wrapper.emitted())
        expect(wrapper.emitted('on:click')).toHaveLength(1)
    })
})