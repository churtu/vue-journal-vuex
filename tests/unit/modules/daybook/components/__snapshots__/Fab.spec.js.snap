// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab component Debe hacer match con el snapshot 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
